### PR TITLE
Start on a Node wrapper for the Desktop client

### PR DESCRIPTION
**Pull request by jrose-signal**
Includes rust/bridge/node/ for the Rust side and node/ for the Electron side. I went with "node" rather than "js" or "ts" because that's kind of baked into the implementation; this would not transparently work in a browser. But I'm open to changes.

### CODE REVIEW

File: node/test/PublicAPITest.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//
+
+import { assert } from 'chai';
+import * as SignalClient from '../index';
+
+describe('SignalClient', () => {
+  it('can generate and serialize PrivateKeys', () => {
+    const a = new SignalClient.PrivateKey();
+    const b = new SignalClient.PrivateKey();
+    assert.equal(a.serialize().length, 32, 'correct length');
```
Comment (EvanHahn-Signal) [inline node/test/PublicAPITest.ts:13]: If you wish, you can Chai's helper here.

```typescript
assert.lengthOf(a.serialize(), 32);
```

What you have is fine.

**Review by EvanHahn-Signal — APPROVED** [review]
Excited for this! Left a few drive-by comments.

File: node/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// For reference: https://github.com/airbnb/javascript
+
+module.exports = {
```
[the remark this answers is not in the document]
Comment (jrose-signal) [reply node/.eslintrc.js:3]: >< I cribbed all the lint setup from RingRTC and then saw it was still using TSLint. I should just take the main repo's ESLint setup.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
```
[the remark this answers is not in the document]
Comment (jrose-signal) [reply package.json:1]: I don't know the significance of this, so we might as well start private!

File: package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "name": "signal-client",
+  "version": "0.1.0",
+  "main": "node/dist/index.js",
+  "types": "node/dist/index.d.ts",
+  "scripts": {
+    "build": "b() { yarn electron-build-env node-gyp configure && yarn electron-build-env node-gyp build ${@:- -d} && yarn tsc; }; b",
```
[the remark this answers is not in the document]
Comment (jrose-signal) [reply package.json:7]: It's a pattern for passing arguments to a particular command in the sequence. This allows saying `yarn build` to get a debug build and `yarn build --release` to get a release build.

File: bin/build-ffi [hunk 5]
```diff
@@ -5,7 +5,10 @@
 # SPDX-License-Identifier: AGPL-3.0-only
 #
 
+set -e
```
[the remark this answers is not in the document]
Comment (jrose-signal) [reply bin/build-ffi:8]: I feel like I had some reason to skip `pipefail` but I'll have to go back through to see what it was. I just forgot about `-u` though!

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrose-signal) [reply package.json:1]: Yeah, private is definitely a reasonable default.

**Pull request merged**